### ISSUE TITLE
Make app wireframes

### ISSUE THREAD

**Issue by dsummersl** [CitizenRedistrictingNC/nc-redistricting#2]

**Comment by dsummersl** [comment]
[Redistricting - 20160620_225614.pdf](https://github.com/CitizenRedistrictingNC/nc-redistricting/files/326484/Redistricting.-.20160620_225614.pdf)

**Comment by garymander** [comment]
This is an excellent start. I have the following comments/questions:
- "Design" Redistricting Plan tracks the language of the bill and may be more descriptive than "Add".
- I think a user should only be able to design a plan if he is logged in.
- The bill only permits "citizens" of NC to submit plans. I don't think we want or need for our purposes to tackle the issue of whether only US citizens can submit plans, but we do need to address the registrant's residence. I would include a box for "Residence Address" and include a statement below the box that "Only North Carolina residents are eligible to submit redistricting plans."
- What is the difference between "Listings" and "Rankings"?
- We don't need to resolve this point now, but eventually we'll have to decide for our redistricting competition how many maps an individual user can submit. Under the bill, there is a charge of $100 for each map submitted. We don't intend to charge an entry fee for our contest, but I don't think we should have a situation where a single individual is submitting dozens of maps. Do others think this is a concern?

**Comment by dsummersl** [comment]
@garymander thanks for the comments, I'll post an updated doc here soon with your feedback.
- Listings is Rankings - oops!
- I don't see a user submitting dozens of maps as a problem unless it is unfair in some way. Is the fear that the leaderboard could have the top ten entries all from the same user?

**Comment by garymander** [comment]
Thanks, Dane. My concern as far as the contest goes is that the winner could end up being the person who has the most time to keep submitting modified maps, although perhaps that is the sort of obsessive behavior we want to encourage. I have no such worries about the actual redistricting procedure under S1093 -- the more maps submitted, the better!

Gary

**Comment by garymander** [comment]
Dane: I realize I’m jumping ahead a bit, but we will obviously need a “Scoring Standards” page on the site. I think these should be taken pretty much verbatim from S1093. I would suggest the following –

Mandatory Requirements

The following are mandatory requirements for all proposed redistricting plans:

(a)    Each district shall consist of contiguous territory. Point-contiguity shall not be considered contiguous.

(b)   One and only one United States Representative shall represent each district. [Because our program will be limited at this stage to congressional districts, we won’t address state senators and representatives]

(c)    The districts shall comply with federal law. [Additional discussion regarding Voting Rights Act to follow]

Scoring Criteria

```
        [Insert text of Section 26, Paragraph 12(a), (b), and (c) from S1093]
```

Please let me know your thoughts. Thanks.

Gary

**Comment by dsummersl** [comment]
Ah, thanks for catching that. I'll be sure to add a mockup for that. This sounds like the kind of thing that we might want to break into two parts:
- A notification section that we display or provide a link to...on the 'Add Redistricting Plan' and 'District Score' pages.
- It isn't too clear how the scoring works to me as yet, but if it's possible to generate maps that don't conform to the three criteria then we should include them in the scoring. Say a checkbox when the three requirements are met.

**Comment by garymander** [comment]
Dane: I hope you had a good 4th. I’ve been giving some more thought to your recent comments.

Regarding compliance with the 3 mandatory standards, the 2nd standard should be easy, as we are not permitting multi-member districts.

The 3rd standard will likely come down to compliance with the Voting Rights Act. As I indicated, I expect this to be a matter of determining how many “majority-minority” districts will be required. When we get to this point (after our initial demo), we will probably want to require the contestants to identify which of the districts shown on their map qualify.

I think checking a box should be sufficient for satisfying the contiguity standard. It should be obvious whether the districts are contiguous. Once a map has been scored, we may have to provide for a visual inspection before it is posted on our site just to make sure there is no contiguity issue.

As to the general topic of scoring, I am hopeful that Auto-Redistrict will give us a leg up, as automated scoring is already built into the program. I recognize that the “geometry” criteria in that program (as previously noted, we are not using the “equality” criteria as these are not in our bill) are not identical to ours, but hopefully they are pretty close and will not require too much additional work.

Thanks for all you’re doing, Dane. Let me know when you need additional input.

Gary

**Comment by garymander** [comment]
Dane and Joe: With apologies if I’m stating the obvious, I think the benchmark we should seek to achieve first is demonstrating online, in real time, the drawing of a map containing 13 congressional districts and the scoring of that map according to the 3 standards laid out in the bill. That is the critical proof of concept we need in order to begin enlisting sponsors and supporters. The other elements of the site - creating individual accounts, modifying previously drawn maps, posting leading submissions - are certainly important, but I see these as secondary milestones.

I recognize that it’s easy for me, as the non-technical guy on this team, to say where we ought to go on software development, as I admittedly don’t have the ability to take us there. But I want to be sure we’re giving you the clearest possible guidance for the invaluable work you’re doing. Please let me know if you’re not getting the input you need.

I can’t tell you how much we appreciate your contributions to this potentially groundbreaking reform effort. Many thanks.

Gary

**Comment by garymander** [comment]
I agree with that—definitely should start with picking points and generating map.

From: Gary Cole [mailto:theateramsouth@nc.rr.com]
Sent: Sunday, July 10, 2016 3:28 PM
To: 'CitizenRedistrictingNC/nc-redistricting' reply@reply.github.com; 'CitizenRedistrictingNC/nc-redistricting' nc-redistricting@noreply.github.com
Cc: 'Mention' mention@noreply.github.com; Joe Emison joe@buildfax.com
Subject: RE: [CitizenRedistrictingNC/nc-redistricting] Make app wireframes (#2)

Dane and Joe: With apologies if I’m stating the obvious, I think the benchmark we should seek to achieve first is demonstrating online, in real time, the drawing of a map containing 13 congressional districts and the scoring of that map according to the 3 standards laid out in the bill. That is the critical proof of concept we need in order to begin enlisting sponsors and supporters. The other elements of the site - creating individual accounts, modifying previously drawn maps, posting leading submissions - are certainly important, but I see these as secondary milestones.

I recognize that it’s easy for me, as the non-technical guy on this team, to say where we ought to go on software development, as I admittedly don’t have the ability to take us there. But I want to be sure we’re giving you the clearest possible guidance for the invaluable work you’re doing. Please let me know if you’re not getting the input you need.

I can’t tell you how much we appreciate your contributions to this potentially groundbreaking reform effort. Many thanks.

Gary

**Comment by dsummersl** [comment]
Hi @garymander, thanks for making clear the priorities. I've been in touch with Chris and we're going to meet on Monday to talk about of the algorithm. I'll post a followup after we do that.

**Comment by garymander** [comment]
That sounds great, Dane. Thanks.

Gary

**Comment by dsummersl** [comment]
Hi all,

Here is an updated doc of mockups. Let me know what you think.

I'll write you separately about my meeting with Chris and plan out the next little milestone.

[Redistricting - 20160620_225614.pdf](https://github.com/CitizenRedistrictingNC/nc-redistricting/files/377211/Redistricting.-.20160620_225614.pdf)

**Comment by garymander** [comment]
Dane: Nice work. I have the following questions/comments –

```
        + If use of the ESRI program will dramatically reduce the time to generate a map, as you indicate, then I agree it makes sense to see what can be worked out with them. Our objective is to demonstrate proof of the S1093 concept through our contest. We are not saying that the state could adopt the citizen redistricting approach without incurring some additional expense, i.e., a license fee to ESRI. It would be nice to avoid if we could, of course, but it may just not be feasible for everything to be open source at this stage.



        + While our focus is on the 13 congressional district map that would be featured in our contest, we need to remember that long term the design approach we take would need to work for maps covering the 50 state Senate seats and 120 state House seats. Do you see Cortney’s anchor approach as scalable to those levels?
```
- When you refer to the automation of Autoredistrict, do you mean that a score for the map would be automatically generated?
  
  ```
      + When someone is designing a map using the approach you describe, will they be able to see in real time the population figures for each district?
  
  
  
      + Chris’s comment about “majority minority” districts is the same point I had made earlier about compliance with the Voting Rights Act. I plan to discuss this point with Chris this week.
  
  
  
      + A reminder that you use “Listings” on the home page but “Rankings” in subsequent pages.
  
  
  
      + I had also mentioned that we will need to include a “Scoring Standards” page laying out the criteria in S1093.
  
  
  
      + Did you, Chris, and Cortney discuss getting them on GitHub?
  ```

Many thanks, Dane. You’re making great progress.

Gary
